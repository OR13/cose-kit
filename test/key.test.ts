import { COSEKeyToJWK, importCOSEKey } from "../src/key/index.js";

describe('COSE key', () => {
  describe('when parsing a public key', () => {
    const coseKey = Buffer.from('a5200121582065eda5a12577c2bae829437fe338701a10aaa375e1bb5b5de108de439c08551d2258201e52ed75701163f7f9e40ddf9f341b3dc9ba860af7e0ca7ca7e9eecd0084d19c01020278246d65726961646f632e6272616e64796275636b406275636b6c616e642e6578616d706c65', 'hex');
    it('should properly decode to JWK', () => {
      const jwk = COSEKeyToJWK(coseKey);
      expect(jwk).toMatchSnapshot();
    });

    it('should be able to be imported as jwk', () => {
      const keyLike = importCOSEKey(coseKey);
      expect(keyLike).toMatchSnapshot();
    });
  });
  describe('when parsing a private key', () => {
    const coseKey = Buffer.from('a601020278246d65726961646f632e6272616e64796275636b406275636b6c616e642e6578616d706c65200121582065eda5a12577c2bae829437fe338701a10aaa375e1bb5b5de108de439c08551d2258201e52ed75701163f7f9e40ddf9f341b3dc9ba860af7e0ca7ca7e9eecd0084d19c235820aff907c99f9ad3aae6c4cdf21122bce2bd68b5283e6907154ad911840fa208cf', 'hex');

    it('should properly decode to JWK', () => {
      const jwk = COSEKeyToJWK(coseKey);
      expect(jwk).toMatchSnapshot();
    });

    it('should be able to be imported as jwk', () => {
      const keyLike = importCOSEKey(coseKey);
      expect(keyLike).toMatchSnapshot();
    });
  });
})
