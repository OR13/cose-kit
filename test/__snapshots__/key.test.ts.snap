// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COSE key when parsing a private key should be able to be imported as jwk 1`] = `Promise {}`;

exports[`COSE key when parsing a private key should properly decode to JWK 1`] = `
{
  "crv": "P-256",
  "d": "r_kHyZ-a06rmxM3yESK84r1otSg-aQcVStkRhA-iCM8",
  "kid": "meriadoc.brandybuck@buckland.example",
  "kty": "EC",
  "x": "Ze2loSV3wrroKUN_4zhwGhCqo3Xhu1td4QjeQ5wIVR0",
  "y": "HlLtdXARY_f55A3fnzQbPcm6hgr34Mp8p-nuzQCE0Zw",
}
`;

exports[`COSE key when parsing a public key should be able to be imported as jwk 1`] = `Promise {}`;

exports[`COSE key when parsing a public key should properly decode to JWK 1`] = `
{
  "crv": "P-256",
  "kid": "meriadoc.brandybuck@buckland.example",
  "kty": "EC",
  "x": "Ze2loSV3wrroKUN_4zhwGhCqo3Xhu1td4QjeQ5wIVR0",
  "y": "HlLtdXARY_f55A3fnzQbPcm6hgr34Mp8p-nuzQCE0Zw",
}
`;
